### PR TITLE
Precheck to verify the correct license secret exists when using an ent image.

### PR DESCRIPTION
**Pull request by NicoletaPopoviciu**
Changes proposed in this PR:
- Added precheck in the `install` command where we verify the correct license secret exists when using an ent image

How I've tested this PR:
- ran tests locally
- ran ```./consul-k8s install -f nicoleta.yaml``` with different values in the yaml file 

How I expect reviewers to test this PR:
👀 

Checklist:
- [x] Tests added
- [x] CHANGELOG entry added 
  > HashiCorp engineers only, community PRs should not add a changelog entry.
  > Entries should use present tense (e.g. Add support for...)

### CODE REVIEW

**Review by ishustava — COMMENTED** [review]
Looking pretty good, Nicoleta! I left some comments inline.

File: cli/cmd/install/install.go [hunk 1]
```diff
@@ -501,3 +530,17 @@ func validLabel(s string) bool {
...
+
+// checkValidEnterprise checks and validates an enterprise installation.
+func (c *Command) checkValidEnterprise(secretName string, image string) error {
+
+	_, err := c.kubernetes.CoreV1().Secrets(c.flagNamespace).Get(c.Ctx, secretName, metav1.GetOptions{})
+	if err != nil {
+		return fmt.Errorf("error getting the enterprise secret for namespace: %s %s", c.flagNamespace, err)
+	}
+	if !strings.Contains(image, "-ent") {
+		return fmt.Errorf("enterprise image not provided: %s", image)
```
[the remark this answers is not in the document]
Comment (ndhanushkodi) [reply cli/cmd/install/install.go:542]: maybe "Consul image: %s is not an enterprise image. An enterprise Consul image must be provided when enterprise license secret is set: %s"

File: cli/cmd/install/install_test.go [hunk 2]
```diff
@@ -187,3 +187,39 @@ func getInitializedCommand(t *testing.T) *Command {
...
+	}
+	secret2 := &v1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "consul-secret2",
+		},
+	}
+
+	// Enterprise secret and image are valid.
+	c.kubernetes.CoreV1().Secrets("consul").Create(context.Background(), secret, metav1.CreateOptions{})
+	c.kubernetes.CoreV1().Secrets("unrelated").Create(context.Background(), secret, metav1.CreateOptions{})
```
Comment (ndhanushkodi) [inline cli/cmd/install/install_test.go:207]:
```suggestion
	c.kubernetes.CoreV1().Secrets("unrelated").Create(context.Background(), secret2, metav1.CreateOptions{})
```

I think this is supposed to be secret2 for the test case in L221

**Review by ndhanushkodi — APPROVED** [review]
Left a comment about a test, but otherwise I have nothing blocking the merge other than what Iryna mentioned!

**Review by ishustava — APPROVED** [review]

**Pull request merged**